### ISSUE TITLE
question: mimalloc shows significant presense on cpu profiler graph

### ISSUE THREAD

**Issue by romange** [microsoft/mimalloc#669]
I am almost sure it's expected - it happens during a very active memory growth phase. Still, I would like to double-check that there is nothing that can be tuned in order to reduce the CPU usage of these routines. We are using v2.0.5

![image](https://user-images.githubusercontent.com/3674760/208256120-94d0d714-5341-4e1c-b11b-532db508acb5.png)

**Comment by daanx** [comment]
Very interesting (can you tell me how I can generate such graphs myself?).

In a growth phase it is a bit expected to see this of course; I am a bit surprised by the `page_extend` taking so much time... You may go into `src/page.c` and redefine `MI_MAX_EXTEND_SIZE` to be, say, `8*1024` (twice as large as now). Let me know if it makes a difference.

Second, the `_mprotect` calls are quite expensive :-(. this is used to commit the memory so you may experiment by setting `MI_EAGER_COMMIT_DELAY=0` so the segments get committed completely upfront. (this will still allow fine grained decommit/commit later on to battle fragmentation).  Can you see if that makes a difference? Otherwise I am surprised by that many commits; did you change the decommit delay?

**Comment by romange** [comment]
Yes, I use https://github.com/gperftools/gperftools that provides a cpu profile and tcmalloc when linking with your executable. I configure it without tcmalloc functionality and use it as CPU profiler only: https://gperftools.github.io/gperftools/cpuprofile.html

then pprof from https://github.com/google/pprof  can take a prof file generated by cpu profiler can generate such graphs or flamegraphs.  

Another option is to use linux perf tool and create flamegraphs with, say, this tool: 
https://github.com/flamegraph-rs/flamegraph

**Comment by romange** [comment]
we set decommit_delay to zero but in this particular use-case we almost do not have decommits (only growth).

**Comment by daanx** [comment]
I fixed a bug in `dev-slice` that would decommit eagerly committed memory -- thanks! Maybe this will decrease the mprotect percentage? Can you try again with the latest `dev-slice`?
